From 9dc95a03e4a764e42b9e62990bb00feb9307ba63 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@kernel.dk>
Date: Tue, 7 Nov 2023 08:05:52 -0700
Subject: [PATCH] test/no-mmap-inval: 0 return is fine too

Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 test/no-mmap-inval.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/no-mmap-inval.c b/test/no-mmap-inval.c
index e851078..9571fee 100644
--- a/test/no-mmap-inval.c
+++ b/test/no-mmap-inval.c
@@ -31,7 +31,7 @@ int main(int argc, char *argv[])
 	if (ret == -EINVAL) {
 		/*  kernel doesn't support SETUP_NO_MMAP */
 		return T_EXIT_SKIP;
-	} else if (ret != -EFAULT) {
+	} else if (ret && ret != -EFAULT) {
 		fprintf(stderr, "Got %d, wanted -EFAULT\n", ret);
 		return T_EXIT_FAIL;
 	}
-- 
2.43.0

