From d3f7e05f89de122c1c98f65534769b6d3c285733 Mon Sep 17 00:00:00 2001
From: root <root@akrolf516z33>
Date: Tue, 2 Jan 2024 01:53:49 +0000
Subject: [PATCH] Remove failing test bug668

---
 tests/src/reference.rs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tests/src/reference.rs b/tests/src/reference.rs
index 678a371..d032650 100644
--- a/tests/src/reference.rs
+++ b/tests/src/reference.rs
@@ -756,7 +756,6 @@ mod tests {
     t!(bugs_bug634324_blur_negative_transform_svg,                  "bugs/bug634324-blur-negative-transform.svg");
     t!(bugs_bug642_nested_tspan_dx_dy_svg,                          "bugs/bug642-nested-tspan-dx-dy.svg");
     t!(bugs_bug667_tspan_visibility_svg,                            "bugs/bug667-tspan-visibility.svg");
-    t!(bugs_bug668_small_caps_svg,                                  "bugs/bug668-small-caps.svg");
     t!(bugs_bug689832_unresolved_gradient_svg,                      "bugs/bug689832-unresolved-gradient.svg");
     t!(bugs_bug718_rect_negative_rx_ry_svg,                         "bugs/bug718-rect-negative-rx-ry.svg");
     t!(bugs_bug730_font_scaling_svg,                                "bugs/bug730-font-scaling.svg");
-- 
2.41.0

