--- a/tools/scripts/approvalTests.py	2023-12-29 17:12:19.318099137 +0000
+++ b/tools/scripts/approvalTests.py	2023-12-29 17:35:56.934845373 +0000
@@ -121,35 +121,38 @@
     # strip source line numbers
     # Note that this parser assumes an already normalized filepath from above,
     # and might break terribly if it is moved around before the normalization.
-    line = filelocParser.sub('\g<filename>:<line number>', line)
+    line = filelocParser.sub(r'\g<filename>:<line number>', line)
 
-    line = lineNumberParser.sub(" ", line)
+    line = lineNumberParser.sub(r" ", line)
 
     if isCompact:
         line = line.replace(': FAILED', ': failed')
         line = line.replace(': PASSED', ': passed')
 
     # strip out the test order number in TAP to avoid massive diffs for every change
-    line = tapTestNumParser.sub("\g<1> {test-number} -", line)
+    line = tapTestNumParser.sub(r"\g<1> {test-number} -", line)
 
     # strip Catch2 version number
-    line = versionParser.sub("<version>", line)
+    line = versionParser.sub(r"<version>", line)
+
+    # Remove Catch2-<version>/ prefix from files
+    line = line.replace('Catch2-<version>/', '')
 
     # replace *null* with 0
-    line = nullParser.sub("0", line)
+    line = nullParser.sub(r"0", line)
 
     # strip executable name
-    line = exeNameParser.sub("<exe-name>", line)
+    line = exeNameParser.sub(r"<exe-name>", line)
 
     # strip hexadecimal numbers (presumably pointer values)
-    line = hexParser.sub("0x<hex digits>", line)
+    line = hexParser.sub(r"0x<hex digits>", line)
 
     # strip durations and timestamps
-    line = junitDurationsParser.sub(' time="{duration}"', line)
-    line = durationParser.sub(' duration="{duration}"', line)
-    line = timestampsParser.sub('{iso8601-timestamp}', line)
-    line = specialCaseParser.sub('file:\g<1>', line)
-    line = sinceEpochParser.sub('{since-epoch-report}', line)
+    line = junitDurationsParser.sub(r' time="{duration}"', line)
+    line = durationParser.sub(r' duration="{duration}"', line)
+    line = timestampsParser.sub(r'{iso8601-timestamp}', line)
+    line = specialCaseParser.sub(r'file:\g<1>', line)
+    line = sinceEpochParser.sub(r'{since-epoch-report}', line)
     return line
 
 
